include/deal.II/numerics/vector_tools.templates.h

```diff
@@ -37,6 +38,9 @@
 #include <deal.II/lac/vector_memory.h>
 #include <deal.II/lac/filtered_matrix.h>
 #include <deal.II/lac/constraint_matrix.h>
+#include <deal.II/matrix_free/matrix_free.h>
+#include <deal.II/matrix_free/operators.h>
+#include <deal.II/matrix_free/fe_evaluation.h>
```

Comment:
It is a pity that we need to include the *huge* file `fe_evaluation.h` inside the *huge* file `vector_tools.templates.h`. But we cannot really split it because we promise the user to let him/her include that single file in case an additional instantiation is needed. (Definitely not something for this PR to address.)